### DISCUSSION TITLE
Can I modularize the collections

### DISCUSSION THREAD

**Discussion by onweru** [decaporg/decap-cms#6713] [Q&A] [answered]
I'm configuring a relatively large website with at least a dozen collections, some of which have hundreds of fields. My config.yml file is now running into thousands of lines.

Is there a way that these collections can be defined in their own yaml/json files and then linked on the config file? For maintainability.

**Comment by talves** [comment, marked as answer]
As long as you can pass on the required config.yml or if you create your own build you could use a json file. I do this all the time.

## Combine and create the `yml` file

- Create the json or code equivalent of the config and use a (json or javascript object) to yaml converter

## Using a json config

- Make a project that builds and bundles up your config into the build

Although this isn't documented clearly, it's definitely a great option.

**Reply by onweru** [reply]
@talves, good suggestion, I will give it a try.
